pkg/task/task.go

```diff
@@ -861,22 +864,49 @@ func (m *taskExecutor) getOEV1alpha1SP() (err error) {
 
 // getExtnV1B1Deployment will get the Deployment as specified in the RunTask
 func (m *taskExecutor) getExtnV1B1Deployment() (err error) {
-	deploy, err := m.getK8sClient().GetExtnV1B1DeploymentAsRaw(m.getTaskObjectName())
+	dclient := deployment.KubeClient(deployment.WithKubeClient(m.getK8sClient().GetKCS()))
+	d, err := dclient.GetExtnV1Beta1(m.getTaskObjectName(), m.getTaskRunNamespace())
 	if err != nil {
 		return
 	}
-	util.SetNestedField(m.templateValues, deploy, string(v1alpha1.CurrentJSONResultTLP))
+
+	util.SetNestedField(m.templateValues, d, string(v1alpha1.CurrentJSONResultTLP))
 	return
 }
 
-// getAppsV1B1Deployment will get the Deployment as specified in the RunTask
-func (m *taskExecutor) getAppsV1B1Deployment() (err error) {
-	deploy, err := m.getK8sClient().GetAppsV1B1DeploymentAsRaw(m.getTaskObjectName())
+// extnV1B1DeploymentRollOutStatus generates rollout status for a given deployment from deployment object
+func (m *taskExecutor) extnV1B1DeploymentRollOutStatus() (err error) {
```

Comment:
i will add more comments, in the next PR after existing review comments are incorporated.